### ISSUE TITLE
Aggregate forecasts in log-odds space with extremization

### ISSUE THREAD

**Issue by geemus** [geemus/metaculus-ai-forecasting#112]
## Motivation

Today, multi-forecaster aggregation happens in two places and both treat probabilities as if they live on a linear scale:

- `forecast_stats.rb:22-39` — reports the **median** of binary probabilities and per-option medians for multiple choice.
- `consensus.rb:18-28` — feeds raw probabilities to an LLM for synthesis; the prompt warns about clustering near 50% but the inputs themselves carry the bias.

This is the wrong space. Probabilities are bounded in [0,1] and behave non-linearly near the tails. A median of `{0.05, 0.10, 0.50, 0.55}` is `0.30`, but the geometric mean of odds gives `0.16` — much more honest about what the two informed forecasters know. The standard fix is well established in the forecasting literature (Satopää et al. 2014, "Combining Multiple Probability Predictions Using a Simple Logit Model"): pool in log-odds space and apply a small extremization factor `a > 1` to counteract the conservatism of averaging.

Empirically, geometric-mean-of-odds with `a ≈ 1.5–2.5` beats arithmetic means and medians on Good Judgment Project and Metaculus-style data. This is the largest expected accuracy win in the codebase that doesn't require new data or new model calls.

## Approach

Introduce a single aggregation module used by both stats and consensus. Default to log-odds pooling with extremization; keep median available for diagnostics.

### Binary

For probabilities `p_i`, compute log-odds `l_i = log(p_i / (1 - p_i))`, take the mean, multiply by extremization factor `a`, then sigmoid back:

```
p_agg = sigmoid(a * mean(logit(p_i)))
```

### Multiple choice

Per-option logit pooling does not preserve sum-to-1. Use the softmax form: take `log(p_i)` per option, average across forecasters, scale by `a`, softmax across options:

```
p_agg[k] = softmax_k(a * mean_i(log(p_i[k])))
```

This is the natural generalization of geometric-mean-of-odds to the simplex and preserves normalization automatically.

### Numeric / discrete

Percentile-based forecasts can't be pooled in log-odds directly. Use **quantile averaging** on the inverse CDF: at each percentile (5, 10, ..., 95), take the mean across forecasters. This is equivalent to averaging the CDFs and is well-behaved. Extremization for distributions is more subtle and not a clear win, so leave it off here.

## Concrete steps

1. **Add `lib/aggregation.rb`** with pure functions:
   - `Aggregation.pool_binary(probs, a: 1.5)` → Float
   - `Aggregation.pool_multiple_choice(prob_dicts, a: 1.5)` → Hash
   - `Aggregation.pool_numeric(percentile_arrays)` → Array
   - Each clamps inputs away from {0, 1} (use `[1e-6, 1-1e-6]`) before logit.
2. **Wire into `forecast_stats.rb`**: replace the median computations at lines 22-39 with calls to the new module. Keep median in the printed output as a sanity check (label the pooled value as the headline).
3. **Wire into `consensus.rb`**: compute the pooled estimate first and pass it into the consensus prompt as a "mechanical aggregate baseline." The consensus LLM keeps its current job — synthesizing reasoning — but is now anchored to a calibrated baseline rather than reasoning from raw inputs. Update `lib/prompts.rb` `CONSENSUS_SYSTEM_PROMPT` and `lib/prompt_templates/forecast_consensus.erb` to surface this baseline and require an explicit reason if the LLM moves more than ~10 percentage points (binary) away from it.
4. **Make the extremization factor configurable** via `ENV['EXTREMIZATION_FACTOR']` defaulting to `1.5`. This lets us tune from retro data without code changes.
5. **Tests** (`test/test_aggregation.rb`):
   - Identity: pooling identical forecasts returns that value.
   - Extremization: `a=1` matches geometric-mean-of-odds; `a>1` moves further from 0.5.
   - MC sums to 1 within `1e-9` for arbitrary inputs.
   - Numeric quantile averaging preserves monotonicity.
   - Edge cases: a forecaster reporting 0% or 100% (must be clamped, not crash).

## Files touched

- `lib/aggregation.rb` (new)
- `forecast_stats.rb`
- `consensus.rb`
- `lib/prompts.rb`
- `lib/prompt_templates/forecast_consensus.erb`
- `test/test_aggregation.rb` (new)
- `test/run_tests.rb` (register new test file)

## Risks & tradeoffs

- **Overconfidence from extremization.** Default `a = 1.5` is conservative. Once issue #2 (per-forecaster calibration weights) lands, `a` should be re-tuned against resolved questions before being raised.
- **Numeric pooling is weaker than binary.** Quantile averaging is fine but doesn't get the same lift as logit pooling. Don't oversell the change for numeric questions.
- **Consensus LLM may still override.** If the LLM frequently rejects the mechanical baseline, that's a signal to tighten the prompt or just submit the baseline directly when forecasters agree.

## Definition of done

- All four question types route through `Aggregation` for the headline number.
- Consensus prompt receives and references the mechanical baseline.
- Tests pass, including new aggregation tests.
- A run on test post IDs (`578`, `14333`, `22427`, `38880` from `lib/utility.rb`) produces sensible outputs and is reviewed for drift versus current behavior.
